q2_metadata/normalization/_norm_rules.py

```diff
@@ -0,0 +1,332 @@
+    _check_variable_rules
+        Loops over each rule and check that they are well formatted
+        and either raise a specific issue error or a fill the rules dict.
+    normalize
+        Perform the application of the variables' rules.
+
+    Raises
+    ------
+    IOError : yaml.YAMLError
+        If something went wrong with reading the .yml rule file.
+    ExpectedError
+        If wrong formatting of an "expected" rule.
+    OntologyError
+        If wrong formatting of an "ontology" rule.
+    RemapError
+        If wrong formatting of a "remap" rule.
+    ValidationError
+        If wrong formatting of a "validation" rule.
+    NormalizationError,
+         If wrong formatting of an "normalization" rule.
+    BlankError
+        If wrong formatting of a "blank" rule.
+    MissingError
+        If wrong formatting of a "missing" rule.
+    FormatError
+        If wrong formatting of a "format" rule.
+
+    """
+    def __init__(self, variable_rules_fp: str):
+        """Initialize the class instance for the set of rules associated
+        with a single metadata variable. The instantiated class will be
+        given a `variable` attribute corresponding to the variable name
+        as string, as well as a `rules` attribute collecting the actual
+        rules in a dictionary.
+
+        Parameters
+        ----------
+        variable_rules_fp : str
+            Path to one variable's rules yaml file.
+
+        """
+        self.rules = {
+            'edits': {
+                'remap': {},
+                'normalization': {},
+                'validation': {}
+            },
+            'lookups': {
+                'ontology': None,
+                'expected': []
+            },
+            'allowed': {
+                'blank': None,
+                'missing': None
+            },
+            'format': None
+        }
+        self.parsed_rules = self.parse_rule(variable_rules_fp)
+        self.warnings = WarningsCollection()
+        self.errors = ErrorsCollection()
+
+    @staticmethod
+    def parse_rule(variable_rules_fp: str):
+        """Read the yaml file of the current variable's rules.
+
+        Parameters
+        ----------
+        variable_rules_fp : str
+            Path to one variable's rules yaml file.
+
+        Returns
+        -------
+        parsed_rules : dict
+            Parsed variable's rules.
+
+        """
+        with open(variable_rules_fp) as handle:
+            try:
+                parsed_rules = yaml.load(handle, Loader=yaml.FullLoader)
+                return parsed_rules
+            except yaml.YAMLError:
+                raise IOError("Something is wrong with the .yml rule file.")
+            except yaml.scanner.ScannerError:
+                raise IOError("Something is wrong with the .yml rule file.")
+            except UnicodeDecodeError:
+                raise IOError("Something is wrong with the .yml rule file.")
```

Comment:
```suggestion
            parsed_rules = yaml.load(handle, Loader=yaml.FullLoader)
            return parsed_rules
```

Reply:
accounted for  - will be part of upcoming commit